doc/app_config.md

```diff
@@ -50,3 +56,100 @@ section-1: value-1  # hasn't been changed
 section-2:
   key-21: value-21-new  # body was replaced
 ```
+
+## Config uploading
+
+In contrast to using `cartridge_app_config`, `cartridge_app_config_path`
+allows loading the config not in sections, but in its entirety.
+To do this, you should specify the path to the configuration file or folder.
+
+An example of indicating the path to different types of config:
+
+```yaml
+cartridge_app_config_path: '../my_app_config.yml'
+# OR
+cartridge_app_config_path: '../my_app_config.zip'
+# OR
+cartridge_app_config_path: '../my_app_config_dir'
+```
+
+There are 3 modes for config loading:
+- `lua` - config will be uploaded by Lua-function `cartridge.config_patch_clusterwide()`;
+- `http` - config will be uploaded to `http://localhost:port/admin/config`
+  with basic authentication by cluster cookie;
+- `tdg` - config will be loaded by a Lua-function in TDG with a version `1.6.15`+, `1.7.6`+ or `2.1.0`+;
+  config will be loaded by HTTP in an older version.
+
+To select any of these types, you should use the variable `cartridge_app_config_upload_mode`:
+```yaml
+cartridge_app_config_upload_mode: 'lua'
+# OR
+cartridge_app_config_upload_mode: 'http'
+# OR
+cartridge_app_config_upload_mode: 'tdg'
+```
+
+### Default modes
+
+To upload file config, you can use Lua-function or HTTP endpoint.
+By default, Lua-function is used to upload YAML files,
+and HTTP endpoint is used to upload other file types.
+
+To upload directory config, you can use HTTP or TDG upload mode.
+The default upload mode is HTTP.
+
+### Lua mode
+
+To upload file config by Lua, `cartridge.config_patch_clusterwide()` is used.
+New config will be parsed, and based on this, the difference will be calculated.
+This difference will be used to patch by `cartridge.config_patch_clusterwide()`.
+
+### HTTP mode
+
+If you use HTTP mode to upload a file, then the file will be sent as is
+to `http://localhost:port/admin/config`. Also, in this mode will be added
+the `Authorization` header for basic authorization using a cluster cookie.
+
+If you use HTTP mode to upload a directory, it will be pre-zipped into a ZIP archive.
+This ZIP archive will be uploaded to `http://localhost:port/admin/config` just like any other file.
+
+The port used to send the config will be taken from the control instance config
+or from the `cartridge_app_config_upload_http_port` variable, if specified.
+
+For example, you can specify the port of the instance,
+which is on the same machine as the control instance:
```

Comment:
"on the same machine as the control" is bad. The user shouldn't know which instance is control, and he shouldn't think about it. I suggest changing `cartridge_app_config_upload_http_port` to `cartridge_app_config_upload_baseurl` to fully delegate to user setting host and port available.